datalad_container/containers_run.py

```diff
@@ -85,14 +85,23 @@ def __call__(cmd, container_name=None, dataset=None,
                 yield res
         assert container, "bug: container should always be defined here"
 
-        image_path = op.relpath(container["path"], pwd)
         # container record would contain path to the (sub)dataset containing
         # it.  If not - take current dataset, as it must be coming from it
-        image_dspath = op.relpath(container.get('parentds', ds.path), pwd)
+        cont_dspath = op.relpath(container.get('parentds', ds.path), pwd)
 
+        image_path = container["path"]
+        # container definition might point to an image in some nested dataset.
+        # it might be useful to be distinguish between the two in such cases
+        image_dspath = op.relpath(get_dataset_root(image_path), pwd)
```

Comment:
`get_dataset_root` is likely not enough here!  We must first ensure that there is a subdataset there, so pretty much perform the same trill as `datalad get` does to install all subdatasets leading to that path.  Will need to check for that later and add a dedicated test for that to have hope to get this PR "proper".